admin/src/js/controllers/edit-user.js

```diff
@@ -254,24 +279,63 @@ angular
       return hasPlace && hasContact;
     };
 
+    const validateFacilityHierarchy = () => {
+      const placeIds = $scope.editUserModel.place;
+
+      if (!placeIds || placeIds.length === 1) {
+        return $q.resolve(true);
+      }
+
+      const getPlace = (placeId) => {
+        return DB().get(placeId);
+      };
+
+      return Promise.all(placeIds.map(getPlace))
+        .then((places) => {
+          const contactTypes = places.map(place => place.contact_type);
+          const isSameHeirarchy = contactTypes.every(type => type === contactTypes[0]);
+
+          if (!isSameHeirarchy) {
+            $translate('permission.description.can_have_multiple_places.incompatible_place').then(value => {
+              $scope.errors.multiFacility = value;
+            });
+          }
+          return isSameHeirarchy;
+        })
+        .catch(err => {
+          $log.error('Error validating facility hierarchy', err);
+          return false;
+        });
+    };
+
     const validateContactIsInPlace = () => {
-      const placeId = $scope.editUserModel.place;
+      const placeIds = $scope.editUserModel.place;
       const contactId = $scope.editUserModel.contact;
-      if (!placeId || !contactId) {
+      if (!placeIds || !contactId) {
         return $q.resolve(true);
       }
-      return DB()
-        .get(contactId)
-        .then(function(contact) {
-          let parent = contact.parent;
-          let valid = false;
-          while (parent) {
-            if (parent._id === placeId) {
-              valid = true;
-              break;
-            }
-            parent = parent.parent;
-          }
+
+      const getParent = (contactId) => {
+        return DB()
+          .get(contactId)
+          .then(function (contact) {
+            return contact.parent;
+          });
+      };
+
+      const checkParent = (parent, placeIds) => {
+        if (!parent) {
+          return false;
+        }
+        if (placeIds.includes(parent._id)) {
+          return true;
+        }
+        return checkParent(parent.parent, placeIds);
+      };
+
+      return getParent(contactId)
+        .then(function (parent) {
+          const valid = placeIds.some((placeId) => checkParent(parent, [placeId]));
```

Comment:
I think you _should_ be able to just call this function by passing all placeIds, right?
```suggestion
const valid = checkParent(parent, placeIds);
```